### ISSUE TITLE
Please provide demo app without docker overhead

### ISSUE THREAD

**Issue by symfonysos** [symlex/symlex#7]
This looks very interesting! However, t would be much easier to get into this framework if I could run the demo just from a local dev server without docker, sqlite instead of mysql and no redis. It woud be great to have such a demo that is easy to use without infrastructure gymnastics!

**Comment by lastzero** [comment]
You can install / use it like any PHP Symfony app using your existing server! :)

However, I think it would be confusion to provide lots of different config files for all possible environments? They are just YAML and configure common components like Doctrine DBAL. It's all documented. `docker-compose up` is actually not very difficult ;) But I know how it feels if you're normally not using Docker... it makes things easier at the end of the day... it's the same discussion in every project.

**Comment by symfonysos** [comment]
Hi, I tried using it like any PHP Symfony app, but `app/console server:run` 
does not exist. 

Regarding docker: docker is a great tool for complex deployment scenarios, but it is a very bad idea to install random docker containers from the internet - do not do that!  Using docker in a professional environment is ok, but of course only with our own registry and hand built images. 

If your demo app can run with only php and sqlite, why add more infrastructure complexity to it? You see the result: instead of taking a quick view into this interesting framework I am postponing everything until day X when I will have time to undockerize that nice demo app, what probably will never happen. Why this roadblocks?

The most simple thing would be a PHP application that _just works_ out of he box, like the symfony demo app. Do not overcomplicate things, keep it simple.

**Comment by lastzero** [comment]
Ah, I see. `app/console server:run` is currently not implemented, you could probably add it, if you don't want to use nginx. Most code can be copied from Symfony I guess. In fact, since I use Docker, I don't need the PHP built-in Web server for anything. To use sqlite, you have to change the config of Doctrine DBAL accordingly, should not be difficult. Also, I don't recommend running real PHP sites with the built-in server and sqlite. For development, I recommend to have a system very similar to what you use in production. Just my 2ct.

Since I'm giving this away for free and provide free support, I don't have time or money to implement everything that people might find useful for them (my day only has 24 hours). I totally understand that other people have other needs and opinions. Pull requests are welcome 👍

**Comment by symfonysos** [comment]
I now just tried to test the demo app with nginx + php-fpm and  get this error when trying to load `web/app.php`: 
```
 PHP Fatal error:  Uncaught ReflectionException: Class Redis does not exist in /home/test/symlex/symlex-demo/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:1127
```

seems like some redis library is missing, I `composer required predis/predis` but that did not help. 

I do not want you to implement things - I just wanted to take a look at this interesting project and I still think it would be a god idea if the demo app just worked. Please do not feel pressured to implement things that you do not need, there are way to many things on everyones todo list. 

I totally understand that resources for supporting a new framework project is something that has to be seriously planned and can not be done by one person alone. Anyway, keep up the good work and good look!

**Comment by lastzero** [comment]
Thank you! :)

Redis is actually a native PHP module, not something you can install with composer. It's for storing the sessions. If you store them in files, it's not scalable and slow since you also get locking effects. As mentioned, I usually work on large, commercial projects so the config is optimized for that. Maybe too much if you just want to play around on your local computer, but the other way (expecting the users to find the right config for production) would be way more difficult.

**Comment by lastzero** [comment]
One of the reasons I recommend Docker. It's just so much easier, especially if you work with different projects that have different requirements (PHP versions, modules,...).

**Comment by lastzero** [comment]
I'll close this now. We just released 4.4.1 in case you want to take a look. Comes with RoadRunner which is much faster in production than NGINX or Apache.
